### DISCUSSION TITLE
opengl getProcAddress in c++?

### DISCUSSION THREAD

**Discussion by hellozyemlya** [slint-ui/slint#5761] [Q&A]
Is that intended that getProcAddress is missing in c++? Also, how to compile and run opengl examples in c++?

**Comment by tronical** [comment]
With C++ the assumption is that you're linking against your EGL/WGL/CGL library and don't need getProcAddress.

The opengl_underlay C++ example requires GLES 2.0 and opengl_texture GLES 3.0. This is checked for here: https://github.com/slint-ui/slint/blob/046b64afec828fbde31c6a0713b87f96f9f4d2e6/examples/CMakeLists.txt#L34

If you configure a clone of the Slint repo and pass `-DSLINT_BUILD_EXAMPLES=ON` they should be compiled, provided that the configure step succeeded in finding GLES2/GLES3 APIs on your system.

**Reply by hellozyemlya** [reply]
Oh, that is great pointer. I was smart enough to use wglGetProcAddress on windows to make sample working fine, but not smart enough to get GLES 2.0 headers and lib. Can you point me to something that can provide that lib and headers for windows and for mac os? I tried angle from vcpkg, but that miserably failing creating shader or program or whatever(I believe it requires some setup as well before it can be used).

Anyway, many thanks, now I can see example up and running.

**Reply by hellozyemlya** [reply]
And another question - why its build only for FEMTOVG? I guess because it does not work with skia-opengl(at least with my wglGetProcAddress hack it does not work with skia, works on femtovg). And why\how then it works with skia-opengl in rust?

**Reply by tronical** [reply]
Angle should work. At least in binary form `libEGL.dll` and `libGLESv2.dll` have worked as drop-in replacement for me in the past. But I'm not sure about the headers. Some how you might have to force the creation of the GL context as EGL instead of WGL.

The FemtoVG guard seems like a mistake, it should also work with Skia's OpenGL renderer (at least it did on my Linux laptop).

**Reply by hellozyemlya** [reply]
Hmm, angle(vcpkg variant) did not worked for me on windows. createprogram returns zero...
